File: ios/wispr-alt/StyleStore.swift
import Foundation

/// Available cleanup styles. Must match the `Style` union in
/// backend/src/transcribe.ts.
enum DictationStyle: String, CaseIterable, Identifiable {
    case clean
    case business
    case casual
    case brief
    case telegram
    case email
    case task

    var id: String { rawValue }

    var label: String {
        switch self {
        case .clean:     return "Чистка"
        case .business:  return "Деловой"
        case .casual:    return "Неформальный"
        case .brief:     return "Краткий"
        case .telegram:  return "Telegram"
        case .email:     return "Email"
        case .task:      return "Задача"
        }
    }

    var hint: String {
        switch self {
        case .clean:    return "Снять «эээ», расставить пунктуацию, разбить на абзацы"
        case .business: return "Формальный рабочий тон, активный залог, структура"
        case .casual:   return "Сохранить разговорный тон, мягкая чистка"
        case .brief:    return "Только суть, маркированные пункты"
        case .telegram: return "Структурированный пост: крючок, абзацы, без эмодзи"
        case .email:    return "Письмо с приветствием, темами и подписью"
        case .task:     return "Action-item: контекст, что сделать, срок"
        }
    }
}

/// Persists the user's preferred dictation style across launches.
@MainActor
final class StyleStore: ObservableObject {
    private static let key = "preferredDictationStyle"

    @Published var current: DictationStyle {
        didSet {
            UserDefaults.standard.set(current.rawValue, forKey: Self.key)
        }
    }

    init() {
        if let raw = UserDefaults.standard.string(forKey: Self.key),
           let style = DictationStyle(rawValue: raw) {
            self.current = style
        } else {
            self.current = .clean
        }
    }
}

File: android/app/src/main/java/com/wispralt/keyboard/StyleStore.kt
package com.wispralt.keyboard

import android.content.Context

/**
 * Available cleanup styles. Must mirror the `Style` union in
 * backend/src/transcribe.ts AND iOS DictationStyle.
 */
enum class DictationStyle(
    val raw: String,
    val label: String,
    val hint: String,
) {
    CLEAN(    "clean",    "Чистка",       "Снять «эээ», расставить пунктуацию, разбить на абзацы"),
    BUSINESS( "business", "Деловой",      "Формальный рабочий тон, активный залог, структура"),
    CASUAL(   "casual",   "Неформальный", "Сохранить разговорный тон, мягкая чистка"),
    BRIEF(    "brief",    "Краткий",      "Только суть, маркированные пункты"),
    TELEGRAM( "telegram", "Telegram",     "Структурированный пост: крючок, абзацы, без эмодзи"),
    EMAIL(    "email",    "Email",        "Письмо с приветствием, темами и подписью"),
    TASK(     "task",     "Задача",       "Action-item: контекст, что сделать, срок"),
    ;

    companion object {
        fun fromRaw(raw: String?): DictationStyle =
            entries.firstOrNull { it.raw == raw } ?: CLEAN
    }
}

/** Persists the user's preferred dictation style in SharedPreferences. */
object StyleStore {
    private const val PREFS = "wispr-alt-style"
    private const val KEY = "style"

    fun get(ctx: Context): DictationStyle {
        val raw = ctx.getSharedPreferences(PREFS, Context.MODE_PRIVATE)
            .getString(KEY, DictationStyle.CLEAN.raw)
        return DictationStyle.fromRaw(raw)
    }

    fun set(ctx: Context, style: DictationStyle) {
        ctx.getSharedPreferences(PREFS, Context.MODE_PRIVATE)
            .edit()
            .putString(KEY, style.raw)
            .apply()
    }
}
